x-pack/plugins/fleet/common/services/is_agent_upgradeable.test.ts

```diff
@@ -171,11 +180,49 @@ describe('Fleet - isAgentUpgradeable', () => {
       isAgentUpgradeable(getAgent({ version: '7.9.0', upgradeable: true }), '8.0.0', '7.9.0')
     ).toBe(false);
   });
-  it('returns false if agent reports upgradeable, but is already updating', () => {
+  it('returns false if agent with no upgrade details reports upgradeable, but is already upgrading', () => {
+    expect(
+      isAgentUpgradeable(
+        getAgent({ version: '7.9.0', upgradeable: true, upgrading: true }),
+        '8.0.0'
+      )
+    ).toBe(false);
+  });
+  it('returns false if agent reports upgradeable, but has an upgrade status other than failed', () => {
     expect(
-      isAgentUpgradeable(getAgent({ version: '7.9.0', upgradeable: true, updating: true }), '8.0.0')
+      isAgentUpgradeable(
+        getAgent({
+          version: '7.9.0',
+          upgradeable: true,
```

Comment:
Does the agent return `upgradeable: true` when it is upgrading?

Reply:
Good question, I'm not sure it does for all upgrade states. I'll check with Elastic Agent team.

Reply:
Hey @ycombinator Do you have an answer to this question?

Reply:
Sorry @jillguyonnet, this slipped through the cracks.  The `local_metadata.upgradeable` field that Agent sends as part of the check-in request does not change its value based on whether the Agent is upgrading or not.  It's a more static value that can be determined based on how the Agent is built (with a certain build-time flag or not), if it's running after being installed, and if it's running under the control of a supervisor. 

https://github.com/elastic/elastic-agent/blob/26947871957102cd972210202f15217789ff5cb9/internal/pkg/agent/application/info/agent_metadata.go#L169-L171

In other words, this value will be fixed once the Agent starts running and doesn't change after that, regardless of whether the Agent is in the middle of an upgrade or not.  Hope that helps!

Reply:
Thank you @ycombinator, that's good to know! Then it seems quite likely that this value will be `true` for upgrading agents.